src/bubble_sort.py

```diff
@@ -0,0 +1,28 @@
+"""Bubble sort algorithm."""
+import timeit
+
+
+def bubble_sort(l):
+    """Bubble Sort: Input list or tuple."""
+    if not isinstance(l, (list, tuple)):
+        raise TypeError('Please input a list or tuple.')
+    n = len(l)
+    swapped = True
+    while swapped:
+        swapped = False
+        for i in range(n - 1):
+            j = i + 1
```

Comment:
I suppose assigning `i + 1` to `j` here saves you the trouble of writing `i + 1` all the time...